### ISSUE TITLE
Allow not indenting match arms

### ISSUE THREAD

**Issue by ubsan** [rust-lang-nursery/rustfmt#2937]
It's a common style in C++ to see:

```cpp
switch (foo) {
case Foo::Bar: {
  ...
} break;
case Foo::Baz: {
  ...
} break;
}
```

I would like a similar style to be possible with rustfmt:

```rust
match foo {
| Foo::Bar => {
    ...
}
| Foo::Baz => {
    ...
}
}
```

or

```rust
match foo {
Foo::Bar => {
    ...
}
Foo::Baz => {
    ...
}
}
```

This would require two configuration options, probably; `|` before each case, and the non-indentation.

Thanks!

**Comment by myrrlyn** [comment]
I strongly agree with this! Nested matches get especially painful; I have instances in my work where, with `impl/fn` already consuming two indents, a triple `match` level adds on another six and I don't get to write actual code until 64 columns in

My personal possibly bad habits about code layout aside, indenting match arm conditions *and* bodies feels redundant and unnecessarily wasteful to me; they don't *both* add meaning to rightward drift.

I think a `| arm` would provide a visual half-indent and be a nice blend of keeping the interior of the match body indented without costing additional space.

**Comment by effinsky** [comment]
hey how is this thing going? would love that as well... especially since you can already do pipes in front of an arm to make it more recognizable. these "indent penalties" are killing me for nesting matches, and it's a wonderful feature and they shouldn't.
